components/atoms/Text/Text.tsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react'
+import {
+  StyleProp,
+  StyleSheet,
+  Text as NativeText,
+  TextStyle,
+} from 'react-native'
+import useTheme from '../../hooks/useTheme'
+
+export interface TextProps {
+  /** Text's Content */
+  children?: React.ReactNode
+  /** And object of additional Styles
+   *
+   * eg: { fontSize: 24 }
+   */
+  style?: StyleProp<TextStyle>
+}
+
+const Text: React.FC<TextProps> = (props) => {
+  const { style = {}, children } = props
+  const theme = useTheme()
+
+  const styles = StyleSheet.create({
+    default: {
+      ...theme.typography.base,
+      color: theme.colors.neutral,
```

Comment:
@J-Rehan I think the order of this is backwards, no? As it's written, I believe the color will always be neutral and you won't be able to overwrite it. In order to have a default but support overwriting, wouldn't it need to be:

```
default: {
   color: colors.neutral,
   ...theme.base
}
```

I could be wrong though

Reply:
The overriding logic is applied here. **[styles.default, `style`]**

https://github.com/explanation/lava-components/blob/9eb15fa12fb1f1369e1042058ddde36bb3ac50d3/components/atoms/Text/Text.tsx#L31